edgelet/edgelet-utils/src/lib.rs

```diff
@@ -70,22 +69,31 @@ pub fn prepare_cert_uri_module(hub_name: &str, device_id: &str, module_id: &str)
     )
 }
 
+const ALLOWED_CHAR_DNS: char = '-';
+const DNS_MAX_SIZE: usize = 63;
+
+/// The name returned from here must conform to following rules (as per RFC 1035):
+///  - length must be <= 63 characters
+///  - must be all lower case alphanumeric characters or '-'
+///  - must start with an alphabet
+///  - must end with an alphanumeric character
+pub fn sanitize_dns_label(name: &str) -> String {
+    name.trim_start_matches(|c: char| !c.is_ascii_alphabetic())
+        .trim_end_matches(|c: char| !c.is_ascii_alphanumeric())
+        .to_lowercase()
+        .chars()
+        .filter(|c| c.is_ascii_alphanumeric() || c == &ALLOWED_CHAR_DNS)
+        .take(DNS_MAX_SIZE)
+        .collect::<String>()
+}
+
 pub fn prepare_dns_san_entries(names: &[&str]) -> String {
     names
         .iter()
-        .map(|name| {
-            // The name returned from here must conform to following rules (as per RFC 1035):
-            //  - length must be <= 63 characters
-            //  - must be all lower case alphanumeric characters or '-'
-            //  - must start with an alphabet
-            //  - must end with an alphanumeric character
-            let name = name
-                .to_lowercase()
-                .trim_start_matches(|c| !char::is_ascii_lowercase(&c))
-                .trim_end_matches(|c| !char::is_alphanumeric(c))
-                .replace(|c| !(char::is_alphanumeric(c) || c == '-'), "");
-
-            format!("DNS:{}", &name[0..cmp::min(name.len(), 63)])
+        .filter_map(|name| {
+            Some(sanitize_dns_label(name))
```

Comment:
While I am all for functional style development, I wonder if in this case the following is clearer:
```rust
let dns = sanitize_dns_label(name);
if dns.is_empty() {
  None
} else {
  Some(format!("DNS:{}", dns))
}
```

Reply:
It is more clear - it's what I had until I tried to find a more terse way to write it. Changing it.